hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/master/BackupLogCleaner.java

```diff
@@ -176,12 +187,29 @@ protected static boolean canDeleteFile(Map<Address, Long> addressToLastBackupMap
       Address walServerAddress = Address.fromString(hostname);
       long walTimestamp = AbstractFSWALProvider.getTimestamp(path.getName());
 
-      if (
-        !addressToLastBackupMap.containsKey(walServerAddress)
-          || addressToLastBackupMap.get(walServerAddress) >= walTimestamp
-      ) {
+      if (!addressToLastBackupMap.containsKey(walServerAddress)) {
+        if (LOG.isDebugEnabled()) {
+          LOG.debug("No backup found for server: {}. Deleting file: {}",
+            walServerAddress.getHostName(), path);
+        }
         return true;
       }
+
+      Long lastBackupTs = addressToLastBackupMap.get(walServerAddress);
+      if (lastBackupTs > walTimestamp) {
+        if (LOG.isDebugEnabled()) {
+          LOG.debug(
+            "Backup found for server: {}. Backup from {} is newer than file, so deleting: {}",
+            walServerAddress.getHostName(), lastBackupTs, path);
+        }
+        return true;
```

Comment:
This is not a debugging refactor -- this new if-block introduces new cleaning logic, yes?

Reply:
This block matches line 181 of the original. Looks fine to me.

Reply:
Correct, outside of [the typo](https://github.com/apache/hbase/pull/6195/files#r1743256609) that Nihal spotted, this should be the same logic but broken up into multiple if-blocks to support explicit debugging logging per-case

Reply:
Got it, thanks.